v2/cmd/voucher_client/config.go

```diff
@@ -23,9 +23,13 @@ func getCheck() string {
 }
 
 func getVoucherClient() (voucher.Interface, error) {
-	newClient, err := client.NewClient(defaultConfig.Server)
-	if nil == err {
-		newClient.SetBasicAuth(defaultConfig.Username, defaultConfig.Password)
+	var newClient *client.Client
+	newClient, err := client.NewAuthClient(defaultConfig.Server)
+	if err != nil {
```

Comment:
FYI, this will not fail in GKE, so this will be executed only if voucher runs in a non-GCP environment. You need to use `NewClient` and `SetBasicAuth` directly, without `AuthClient` for it to work in GKE.

Reply:
I wonder if we should have some flag to indicate that auth is required, so if I want to use the voucher_client cli I can specify that this requires auth via id tokens. Then, we can call either `NewClient` or `AuthClient` depending on auth being enabled or not...

Reply:
That could be another option for sure. I was considering that, but I thought it'd be cleaner to choose between 2 functions instead of providing a flag or a boolean.

Reply:
Yeah. My thought here was that we wouldn't be getting an error back and it would try to create a client with Auth and it would fail the vouch checks and we've see in `auto-voucher`. With the flag approach, we're making sure it's backwards compatible, so we know for sure which client o instantiate based on the flag